spec/units/line_editor_spec.rb

```diff
@@ -0,0 +1,702 @@
+require 'spec_helper'
```

Comment:
It's mostly a port of the Ruby test suite's Readline tests, converted from minitest to RSpec, and better organised by method.

I intend to do more refactoring here before this is merged.